CSharpSourceCode/Battle/TriggeredEffect/TriggeredEffect.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Xml.Serialization;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+using TaleWorlds.Engine;
+using System.Timers;
+using System;
+
+namespace TOW_Core.Battle.TriggeredEffect
+{
+    public class TriggeredEffect : IDisposable
+    {
+        private TriggeredEffectTemplate _template;
+        private int _soundIndex;
+        private SoundEvent _sound;
+
+        public TriggeredEffect(TriggeredEffectTemplate template)
+        {
+            _template = template;
+        }
+
+        public void Dispose()
+        {
+            CleanUp();
+        }
+
+        public void Trigger(Vec3 position, Vec3 normal, Agent triggererAgent)
```

Comment:
maybe also make an overload for a single agent target something Like Trigger(Agent target, Agent triggererAgent)